hack/toolchain.sh

```diff
@@ -11,17 +11,17 @@ main() {
 
 tools() {
     go install github.com/google/go-licenses@v1.6.0
-    go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.55.2
-    go install github.com/google/ko@v0.15.1
-    go install github.com/mikefarah/yq/v4@v4.40.5
-    go install github.com/norwoodj/helm-docs/cmd/helm-docs@v1.12.0
-    go install sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.0.0-20231031112659-edb7fe9c520e
+    go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.57.2
+    go install github.com/google/ko@v0.15.2
+    go install github.com/mikefarah/yq/v4@v4.43.1
+    go install github.com/norwoodj/helm-docs/cmd/helm-docs@v1.13.1
+    go install sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.0.0-20240409134613-20f3f4bed925
     go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.14.0
-    go install github.com/sigstore/cosign/v2/cmd/cosign@v2.2.3
+    go install github.com/sigstore/cosign/v2/cmd/cosign@v2.2.4
 #   go install -tags extended github.com/gohugoio/hugo@v0.110.0
-    go install golang.org/x/vuln/cmd/govulncheck@v1.0.3
-    go install github.com/onsi/ginkgo/v2/ginkgo@v2.15.0
-    go install github.com/rhysd/actionlint/cmd/actionlint@v1.6.26
+    go install golang.org/x/vuln/cmd/govulncheck@v1.0.4
+    go install github.com/onsi/ginkgo/v2/ginkgo@latest
```

Comment:
This means we will reinstall the ginkgo CLI whenever there is an update. Currently running `make presubmit` and having it trigger `make toolchain` works because it doesn't often change. 

We don't update it frequently: https://github.com/onsi/ginkgo/releases 

This is still something to consider.

Reply:
Considered. The very reason for changing to `@latest` is because we usually bump the ginkgo packages (e.g. via dependabot) and CLI gets outdated, which generates a warning. So making an exception from pinning the versions here.